CMakeLists.txt

```diff
@@ -13,17 +13,31 @@ find_package(catkin REQUIRED
     sparse_bundle_adjustment
     tf
     visualization_msgs
+    pluginlib
 )
 find_package(Eigen REQUIRED)
 
 include_directories(${catkin_INCLUDE_DIRS} ${EIGEN_INCLUDE_DIRS})
 add_definitions(${EIGEN_DEFINITIONS})
 
-catkin_package()
+catkin_package(
+  LIBRARIES karto_scan_solver_plugins karto_graph_visualizer_plugins
+  )
 
-add_executable(slam_karto src/slam_karto.cpp src/spa_solver.cpp)
-target_link_libraries(slam_karto ${catkin_LIBRARIES})
+add_library(karto_scan_solver_plugins src/spa_solver.cpp)
+add_library(karto_graph_visualizer_plugins src/spa_graph_visualizer.cpp)
+add_executable(slam_karto src/slam_karto.cpp)
 
-install(TARGETS slam_karto
+target_link_libraries(karto_scan_solver_plugins ${catkin_LIBRARIES})
+target_link_libraries(karto_graph_visualizer_plugins ${catkin_LIBRARIES})
+target_link_libraries(slam_karto ${catkin_LIBRARIES} karto_scan_solver_plugins karto_graph_visualizer_plugins)
```

Comment:
@stwirth That's because I use the SPA solver and graph visualizer as fallback behaviors here : https://github.com/alain-m/slam_karto/pull/2/files#diff-90eaf8817c1366730d834fb0bee07234R316  and I don't load them through a class loader. But I can change that.